File: plugin_files/bin/scripting/entity.lua
local hookEntityOutputHandles = {
    pre = {},
    post = {}
}

function AddHookEntityOutput(classname, output)
    if type(classname) ~= "string" then
        print("AddHookEntityOutput: classname must be a string.")
        return
    end
    if type(output) ~= "string" then
        print("AddHookEntityOutput: output must be a string.")
        return
    end

    local hookId = hooks:EntityOutputAdd(classname, output)

    if hookId == "00000000-0000-0000-0000-000000000000" then
        return
    end

    return { hookId = hookId }
end

File: plugin_files/bin/scripting/entity.js
const LoadEntities = (global) => {
    global.AddHookEntityOutput = (classname, output) => {
        if(typeof classname != "string") return console.log("AddHookEntityOutput: classname must be a string.");
        if(typeof output != "string") return console.log("AddHookEntityOutput: output must be a string.");
    
        let hookId = hooks.AddEntityOutputHook(classname, output)
    
        if (hookId == "00000000-0000-0000-0000-000000000000") return
    
        return { hookId }
    }
}

LoadEntities(globalThis)